scripts/uniform.py

```diff
@@ -1,178 +1,78 @@
+    filepath,
+    feature_types,
+    week_type,
+    course,
+    feature_names,
+    remove_obvious=remove_obvious,
+    fill=fill,
+    drop=drop,
+    normalize=normalize,
+)
 SHAPE = features.shape
 
-print(features.shape)
-print('course: ', course)
-print('week_type: ', week_type)
-print('feature_type: ', feature_types)
-print(selected_features)
-# calculate the number of features
-n_features = sum([len(x) for x in selected_features])
-
-# make feature names more readable
-# ex: time_in__problem_<function sum at 0x7f3bd02cc9d0> -> time_in_problem_sum
-def clean_name(feature):
-  id = feature.find('<')
-  if id==-1:
-    return feature
-  fct = feature[id+9:id+14].strip()
-  return feature[0:id]+fct
-
-selected_features = [np.array([clean_name(x) for x in selected_features[0]]),
- np.array([clean_name(x) for x in selected_features[1]]),
- np.array([clean_name(x) for x in selected_features[2]]),
- np.array([clean_name(x) for x in selected_features[3]])]
-
-selected_features={
-    "boroujeni_et_al":list(selected_features[0]),
-     "chen_cui":list(selected_features[1]),
-    "marras_et_al":list(selected_features[2]),
-    "lalle_conati":list(selected_features[3])
-}
-print(selected_features)
-file = 'selected_features/' + course + '_after.json'
-with open(file, 'w') as f: 
-    json.dump(selected_features, f)
-
-num_feature_type = []
-for feature_type in feature_types:
-    num_feature_type.append(len(selected_features[feature_type]))
-print(num_feature_type)
-
-# Loading feature names and transforming them to 2D format.
-feature_names= []
-final_features = []
-for feature_type in feature_types:
-    [final_features.append(x) for x in selected_features[feature_type]]
-for i in np.arange(n_weeks):
-    feature_type_name_with_weeks = [(x+'_InWeek'+str(i+1)) for x in final_features]
-    feature_names.append(feature_type_name_with_weeks)
-feature_names = np.concatenate(feature_names, axis=0)
-feature_names = feature_names.reshape(-1)
-# print(feature_names)
-features.columns = feature_names
-
-
-# ## Making a predict_proba
-features_min = features.min(axis=0)
-features_max = features.max(axis=0)
-features_max = np.where(features_max==0, np.ones(features_max.shape),features_max)
-
-# This module transforms our data to the 2D format biLSTM was trained with.
-def transform_x(x, num_feature_type, num_weeks, features_min, features_max, normal=True):
-    return np.array(x).reshape((x.shape[0],x.shape[1]))
-
-features_min = features.min(axis=0)
-features_max = features.max(axis=0)
-
+# Loading the labels
+filepath = (
+    "../data/"
+    + week_type
+    + "-"
+    + feature_types[0]
+    + "-"
+    + course
+    + "/feature_labels.csv"
+)
+labels = load_labels(filepath, normalize=normalize)
+# Adding week number to feature_names
+feature_names = add_week(selected_features, feature_types, num_weeks, week_type)
+features = pd.DataFrame(features, columns=feature_names)
 # EDIT HERE FOR OTHER MODELS
```

Comment:
Again, we can maybe pass this in as an argument to the file (with a default value as we have)